doc/nrf/doc_styleguide.rst

```diff
@@ -185,31 +236,53 @@ Functions
 
 #. **@return** should be used to describe a return value (for example, "@return The length of ...", "@return The handle").
    There is usually only one return value.
+
+   .. code-block:: c
+      :caption: Return documentation example
+
+        *  @return  Initializer that sets up the pipe, length, and byte array for
+        *           content of the TX data.
+
 #. **@retval** should be used for specific return values (for example, "@retval true", "@retval CONN_ERROR").
    Describe the condition for each of the return values (for example, "If the function completes successfully", "If the connection cannot be established").
    If there is only one retval, add what happens otherwise. Example: "Otherwise, an error code is returned".
+
+    .. code-block:: c
+       :caption: Retval documentation example
+
+         *  @retval 0 If the operation was successful.
+         *            Otherwise, a (negative) error code is returned.
+         *  @retval (-ENOTSUP) Special error code used when the UUID
+         *            of the service does not match the expected UUID.
+
 #. **Do not use @returns**.
    Use **@return** instead.
 
 .. code-block:: c
-   :caption: Function documentation example
-
-	/** @brief Send Boot Keyboard Input Report.
-	 *
-	 *  @param hids_obj  	HIDS instance.
-	 *  @param rep 		Pointer to the report data.
-	 *  @param len 		Length of report data.
-	 *
-	 *  @retval 0 		If the operation was successful.
-         *                      Otherwise, a (negative) error code is returned.
-	 */
-	int hids_boot_kb_inp_rep_send(struct hids *hids_obj, u8_t const *rep,
-					  u16_t len);
+   :caption: Complete function documentation example
+
+    /** @brief Request a random number from the Secure Firmware.
+     *
+     * This function provides a True Random Number from the on-board random number generator.
+     *
+     * @note Currently, the RNG hardware is run each time this function is called. This
+     *       consumes significant time and power.
+     *
+     * @param[out] output  The random number. Must be at least @c len long.
```

Comment:
tbh, this should be @p instead of @c, @p is meant especially for referring to parameters in running text. Nothing serious (same typewriter font output) , but if we are providing an example, we should be extra correct.